framework/src/modules/chain/blocks/blocks.js

```diff
@@ -584,6 +705,130 @@ class Blocks extends EventEmitter {
 			}
 		);
 	}
+
+	/**
+	 * Block IDs are the same ~ Blocks are equal
+	 * @param block
+	 * @returns {*}
+	 * @private
+	 */
+	// eslint-disable-next-line class-methods-use-this
+	_handleSameBlockReceived(block) {
+		this.logger.debug('Block already processed', block.id);
```

Comment:
is this just for log or TODO?
I think we don't need to create a function just for log 😉

Reply:
A mix of both and also for the sake of consistency. I am unsure on how in the function will evolve in the future or of new logic will need to be added but in my opinion, leaving it this way readability on `_forkChoiceTask` increases.